src/components/MapView/Layers/BoundaryLayer/index.tsx

```diff
@@ -28,21 +39,67 @@ function BoundaryLayer({ layer }: { layer: BoundaryLayerProps }) {
 
   const isPrimaryLayer = isPrimaryBoundaryLayer(layer);
 
-  const onClickFunc = (evt: any) => {
+  const onClickShowPopup = (evt: any) => {
+    dispatch(hidePopup());
     const coordinates = evt.lngLat;
     const locationName = getFullLocationName(
       layer.adminLevelNames,
       evt.features[0],
     );
+
     const locationLocalName = getFullLocationName(
       layer.adminLevelLocalNames,
       evt.features[0],
     );
+
     dispatch(showPopup({ coordinates, locationName, locationLocalName }));
+  };
+
+  const onClickFunc = (evt: any) => {
+    const { properties } = evt.features[0];
+
     // send the selection to the map selection layer. No-op if selection mode isn't on.
     dispatch(
       toggleSelectedBoundary(evt.features[0].properties[layer.adminCode]),
     );
+
+    onClickShowPopup(evt);
+
+    const selectedLayerWMS: undefined | WMSLayerProps = selectedLayers.find(
+      l => l.type === 'wms' && l.chartData,
+    ) as WMSLayerProps;
+
+    if (!selectedLayerWMS) {
+      return;
+    }
+
+    const { serverLayerName, title, chartData } = selectedLayerWMS;
+
+    const { levels, url, type: chartType } = chartData!;
+
+    const lowestLevelId = levels[levels.length - 1].id;
+
+    const boundaryProps = levels.reduce(
+      (obj, item) => ({
+        ...obj,
+        [item.id]: {
+          code: properties[item.id],
+          urlPath: item.path,
+          name: properties[item.name],
+        },
+      }),
+      {},
+    );
+
+    const adminBoundaryParams: AdminBoundaryParams = {
+      title,
+      boundaryProps,
+      serverParams: { layerName: serverLayerName, url },
+      chartType,
+    };
+
+    dispatch(setBoundaryParams(adminBoundaryParams));
+    dispatch(updateAdminId(lowestLevelId));
```

Comment:
If all this logic is for the DataViewer, let's either:
1. package it into a function, maybe moving it to a file more relevant to the DataViewer and import it here
2. Group it and add some comments explaining what we are doing and why.